crates/sui-core/src/checkpoints/checkpoint_executor/mod.rs

```diff
@@ -51,7 +51,7 @@ mod metrics;
 pub(crate) mod tests;
 
 const TASKS_PER_CORE: usize = 1;
-const END_OF_EPOCH_BROADCAST_CHANNEL_CAPACITY: usize = 2;
+const END_OF_EPOCH_BROADCAST_CHANNEL_CAPACITY: usize = 100;
```

Comment:
I'm not sure I'm following why this should ever need to be more than 2. Shouldn't updating the committee in qd be blocking for starting the new epoch on the node? If so, then this queue should never have more than 1 end of epoch messages. If not blocking, how do we guarantee epoch consistency between qd and elsewhere?

Reply:
sorry, forgot to explain the change. This is because when a stale node starts to catch up, its reconfig speed may surpass how receivers (e.g. QD) processes it and cause the new committee message to drop from the queue. Thus the node would miss the committee information in some epochs, which could cause some downstream problems.

Reply:
for example, https://github.com/MystenLabs/sui/blob/main/crates/sui-node/src/lib.rs#L616, if a certain epoch committee message is dropped from the queue, we will skip the `self.state.reconfigure(new_committee.committee)` call hence miss a few things such as insert committee info into committee store

Reply:
Right but that's only a problem if we allow reconfig to finish and move on without quorum driver handling the message. This is what I mean by "is this safe?". The committee that quorum driver is using must always match the committee of the locally configured epoch, no?

Reply:
>  if we allow reconfig to finish and move on without quorum driver handling the message

This is more than quorum driver. The code pointer I shared above is on all validators/nodes. The problem is the broadcast channel has no back pressure and will start to drop older messages when consumer is too slow to poll data.

For "is this safe": yes it is, because there's no point in creating a quorum driver in epoch 100 when validators are already in epoch 200. The important thing we make sure is no committee store update is missed, as mentioned [here](https://github.com/MystenLabs/sui/pull/7024#discussion_r1058554569)